### DISCUSSION TITLE
wire:confirm.prompt is asked again even if button replaced and no confirm exists in new button

### DISCUSSION THREAD

**Discussion by jcerdan** [livewire/livewire#7413] [1. Help] [answered]
Hi,

I'm building a site with PHP 8.1 and laravel 10.34.2 / livewire 3.2.2

I have a set of invoices displayed in a table with a @forelse and a wire:key in <tr>

```
@forelse ($invoices as $invoice)
                <tr class="hover:bg-primary-100" wire:key="{{ $invoice->id }}">
...
                     <x-table-cell>
                        @if ($invoice->rectified)
                            <x-button xs color="info" label="rectified" icon="download" wire:click.prevent="download_rectified_invoice({{ $invoice->rectified->id }})" spinner="download_rectified_invoice({{ $invoice->rectified->id }})" />
                        @else
                            <x-button secondary flat label="rectify" wire:click.prevent="rectify_invoice({{ $invoice->id }})" wire:confirm.prompt="Are you sure?\n\nType YES to confirm|YES"/>
                        @endif
                    </x-table-cell>
...

```

first the invoice is not rectified but clicking on 'rectified' and confirming, creates the rectified invoice and then the first button in if/else is displayed with a link to rectified invoice PDF.

The problem is that when clicking download button, the confirmation dialog appears again! and when typing YES, the PDF is downloaded!

I don't really know if this is a bug on livewire or something I make wrong, but it looks like a livewire bug.

Has someone met this problem?

Regards,

**Comment by PhiloNL** [comment, marked as answer]
Seems like a morphing issue. Can you try adding a wire:key to each button e.g. `wire:key="{{ $invoice->id }}-download"` & `wire:key="{{ $invoice->id }}-rectify"`.

**Reply by jcerdan** [reply]
Hi @PhiloNL ,

Yes! that did it.

Will read again the doc about moprhing as it's clear I didn't understood all issues that can cause.

Best regards!
